source/core/index.ts

```diff
@@ -247,13 +249,20 @@ export default class Request extends Duplex implements RequestEvents<Request> {
 			return;
 		}
 
-		if (this.options.signal?.aborted) {
-			this.destroy(new AbortError(this));
-		}
+		if (this.options.signal) {
+			const abort = () => {
+				this.destroy(new AbortError(this));
+			};
 
-		this.options.signal?.addEventListener('abort', () => {
-			this.destroy(new AbortError(this));
-		});
+			if (this.options.signal.aborted) {
+				abort();
```

Comment:
Good catch, there would still be leak for an already aborted signal. Are you sure about the return though because the `if (is.nodeStream(body)) {` branch on line 270 is also not executed anymore? Alternatively the `addEventListener` would go in an `else`.

Reply:
Done